### DISCUSSION TITLE
Communication around important cubes

### DISCUSSION THREAD

**Discussion by ptbrowne** [visualize-admin/visualization-tool#1009] [General]
Hi,

We use automated tests on our cubes and they are regularly broken due to changes in cubes. To remediate this problem, we would like to be notified when there is a change on these cubes. Here is a list of cubes that are included in our tests

It would be important that we get notified in case of any change to these cubes.
Having the description of changes would be helpful.
We would also be interested in the frequency of update for those cubes.

@rdataflow, @ortnever, do you think it would be possible to have those notifications as part of your process ?

Thanks

| number | iri                                                                                      | cube name                                            |
| ------ | ---------------------------------------------------------------------------------------- | ---------------------------------------------------- |
| 1      | <https://environment.ld.admin.ch/foen/ubd0104/11>                                      | "Bathing water quality (with Hierarchie)"@en         |
| 2      | <https://environment.ld.admin.ch/foen/gefahren-waldbrand-warnung/1>                     | "Forest fire danger"@en                              |
| 3      | <https://environment.ld.admin.ch/foen/gefahren-waldbrand-praeventionsmassnahmen-kantone/1>| "Forest fire prevention measures"@en                |
| 4      | <https://environment.ld.admin.ch/foen/ubd000502_sad_01/6>                              | "Greenhouse gas emissions by sector and by gas"@en   |
| 5      | <https://environment.ld.admin.ch/foen/ubd000502_sad_01/7>                              | "Greenhouse gas emissions by sector and by gas"@en   |
| 6      | <https://environment.ld.admin.ch/foen/ubd0066/15>                                      | "Heavy Metal Soil Contamination"@en                  |
| 7      | <https://environment.ld.admin.ch/foen/fab_hierarchy_test13_switzerland_canton_municipality/3>| "Hierarchy test13 - Switzerland Canton Municipality population"@en |
| 8      | <https://environment.ld.admin.ch/foen/nfi/C-96/cube/1>                                  | "NFI: All target values by Conifers and broadleaves"@en |
| 9      | <https://environment.ld.admin.ch/foen/palmer-penguins/18>                               | "Palmer Penguins"@en                                 |
| 10     | <https://environment.ld.admin.ch/foen/ubd003001/14>                                     | "Red List (latest survey, with %)"@en                |
| 11     | <https://environment.ld.admin.ch/foen/UBD003002/5>                                      | "Red Lists (latest survey)"@en                       |
| 12     | <https://culture.ld.admin.ch/sfa/StateAccounts_Office/1>                                 | "State accounts - Office"@en                         |

**Comment by ortnever** [comment]
Salut Patrick, 
pour les jeux de données qui me concernent il s'agit d'anciennes versions publiées uniquement sur INT et qui sont sans cesse modifiées. Ne faudrait-il pas utiliser des cubes sur PROD pour faire vos tests?

**Comment by ptbrowne** [comment]
> Hi Patrick,
for the datasets that concern me, these are old versions published only on INT and which are constantly modified. Shouldn't you use cubes on PROD to do your tests?

I've translated the message for @bprusinowski who does not understand French.

Hey veronique, the problem is that when we develop features, the cubes are never in production yet, and we've had a lot of trouble to have access to production cubes, it's only recently that there have been cubes in production. Personally, I would certainly prefer to test on more stable cubes.

1. I think that we should be given a list of important production cubes spanning all the functionalities of visualize (hierarchies, all kind of dimensions, geographical dimensions, etc...) and then we can write more robust tests.
2. In any case, we will have to sometime test on cubes in integration. For those cases, it would be nice either to have a way to be notificed when a cube is changed (either automatically, or manually), or have access to a changelog of the data in the cube. It would be nice that it becomes a pattern that when a cube is changed / updated, the data maintainer has to provide a list of the changes.

cc @adintegra What's your thinking here ?

**Reply by Rdataflow** [reply]
@ptbrowne as you may know NFI cubes are in prep for PROD. once there we can also test & monitor them... they won't change a lot.

cc @kronmar
